File: backend/app/reservations/schemas.py
"""
Reservations Module — Schemas
"""
from datetime import date

from pydantic import BaseModel

from app.reservations.models import ReservationSource, ReservationStatus
from app.shared.schemas import BaseResponse


class ReservationCreateRequest(BaseModel):
    property_id: str
    listing_id: str | None = None
    confirmation_code: str | None = None
    status: ReservationStatus = ReservationStatus.CONFIRMED
    source: ReservationSource = ReservationSource.MANUAL
    check_in: date
    check_out: date
    nights: int
    guest_name: str | None = None
    guest_email: str | None = None
    guest_phone: str | None = None
    number_of_guests: int = 1
    gross_revenue: float = 0.0
    cleaning_fee: float = 0.0
    platform_fee: float = 0.0
    taxes: float = 0.0
    currency: str = "USD"
    notes: str | None = None


class ReservationResponse(BaseResponse):
    property_id: str
    listing_id: str | None = None
    confirmation_code: str | None = None
    status: ReservationStatus
    source: ReservationSource
    check_in: date
    check_out: date
    nights: int
    guest_name: str | None = None
    guest_email: str | None = None
    number_of_guests: int
    gross_revenue: float
    cleaning_fee: float
    platform_fee: float
    taxes: float
    net_revenue: float
    currency: str
    notes: str | None = None


class ReservationSummary(BaseResponse):
    """Lightweight reservation for lists."""
    property_id: str
    confirmation_code: str | None = None
    status: ReservationStatus
    check_in: date
    check_out: date
    nights: int
    guest_name: str | None = None
    gross_revenue: float
    currency: str

File: backend-rs/src/reservations/schemas.rs
//! Reservation request schemas, mirroring `backend/app/reservations/schemas.py`.

use serde::Deserialize;

fn default_status() -> String {
    "confirmed".into()
}
fn default_source() -> String {
    "manual".into()
}
fn default_guests() -> i64 {
    1
}
fn default_currency() -> String {
    "USD".into()
}

#[derive(Debug, Deserialize)]
pub struct ReservationCreateRequest {
    pub property_id: String,
    pub listing_id: Option<String>,
    pub confirmation_code: Option<String>,
    #[serde(default = "default_status")]
    pub status: String,
    #[serde(default = "default_source")]
    pub source: String,
    pub check_in: String,
    pub check_out: String,
    pub nights: i64,
    pub guest_name: Option<String>,
    pub guest_email: Option<String>,
    pub guest_phone: Option<String>,
    #[serde(default = "default_guests")]
    pub number_of_guests: i64,
    #[serde(default)]
    pub gross_revenue: f64,
    #[serde(default)]
    pub cleaning_fee: f64,
    #[serde(default)]
    pub platform_fee: f64,
    #[serde(default)]
    pub taxes: f64,
    #[serde(default = "default_currency")]
    pub currency: String,
    pub notes: Option<String>,
}

#[derive(Debug, Default, Deserialize)]
pub struct ListParams {
    pub property_id: Option<String>,
    pub skip: Option<i64>,
    pub limit: Option<i64>,
}
